packages/create-catalyst/src/utils/reset-branch-to-ref.ts

```diff
@@ -0,0 +1,9 @@
+import { execSync } from 'node:child_process';
+
+export function resetBranchToRef(projectDir: string, ghRef: string): void {
+  execSync(`git reset --hard ${ghRef}`, {
+    cwd: projectDir,
+    stdio: 'inherit',
+    encoding: 'utf8',
+  });
+}
```

Comment:
As discussed internally, in the unlikely event that a bad `ghRef` is passed, this will throw and kill the CLI process. This will leave the user running the CLI in a state where the repository was cloned successfully, but dependencies wouldn't be installed nor would the environment file be written (see: https://github.com/bigcommerce/catalyst/blob/ad8c86d574474eb5ed18d99265fe4001d267fb5f/packages/create-catalyst/src/commands/create.ts#L257-L266) 

For now, we'll accept this since it's unlikely the `ghRef` provided by OCC will be rejected. That said, we do need to handle errors in a follow up PR.